recipe-server/normandy/recipes/models.py

```diff
@@ -517,19 +550,55 @@ def recipes_used_by(self):
     def __str__(self):
         return self.name
 
+    def canonical_json(self):
+        # Avoid circular import
+        from normandy.recipes.api.v1.serializers import ActionSerializer
+        data = ActionSerializer(self).data
+        return CanonicalJSONRenderer().render(data)
```

Comment:
I tried this, and I ended up causing really weird import errors I couldn't track down, so I left it as is.